python/ctsm/site_and_regional/single_point_case.py

```diff
@@ -0,0 +1,625 @@
+
+    create_datm_at_point:
+        Extract all DATM data at a single point.
+    """
+
+    # pylint: disable=too-many-instance-attributes
+    # the ones we have are useful
+
+    def __init__(
+            self,
+            plat,
+            plon,
+            site_name,
+            create_domain,
+            create_surfdata,
+            create_landuse,
+            create_datm,
+            create_user_mods,
+            dom_pft,
+            pct_pft,
+            num_pft,
+            include_nonveg,
+            uni_snow,
+            cap_saturation,
+            out_dir,
+    ):
+        super().__init__(create_domain, create_surfdata, create_landuse, create_datm,
+                         create_user_mods)
+        self.plat = plat
+        self.plon = plon
+        self.site_name = site_name
+        self.dom_pft = dom_pft
+        self.pct_pft = pct_pft
+        self.num_pft = num_pft
+        self.include_nonveg = include_nonveg
+        self.uni_snow = uni_snow
+        self.cap_saturation = cap_saturation
+        self.out_dir = out_dir
+
+        self.create_tag()
+        self.check_dom_pft ()
+        self.check_nonveg ()
+        self.check_pct_pft ()
+
+    def create_tag(self):
+        """
+        Create a tag for single point which is the site name
+        or the "lon-lat" format if the site name does not exist.
+        """
+        if self.site_name:
+            self.tag = self.site_name
+        else:
+            self.tag = "{}_{}".format(str(self.plon), str(self.plat))
+
+    def check_dom_pft (self):
+        """
+        A function to sanity check values in dom_pft:
+
+        - Compare dom_pft (values if more than one) with num_pft:
+          i.e. If dom_pft is 18 without crop it fails.
+
+        - Check for mixed land-units:
+          If we have more than one dom_pft, they should be in the
+          same range.
+          e.g. If users specified multiple dom_pft, they should be
+          either in :
+            - 1-14 range
+            or
+            - 15-78 range
+            - give an error : mixed land units not possible.
+
+        dom_pft in netcdf: 1-15 which tranlate to 0-14
+        -------------
+        Raises:
+            Error (ArgumentTypeError):
+                If any dom_pft is bigger than 78.
+            Error (ArgumentTypeError):
+                If any dom_pft is less than 1.
+            Error (ArgumentTypeError):
+                If mixed land units are chosen.
+                dom_pft values are both in range of 1-14 and 15-78.
+
+
+        """
+
+        if self.dom_pft is None:
+            logger.warning ("No dominant pft type is chosen. "
+                            "If you want to choose a dominant pft type, please use --dompft flag.")
+        else:
+            min_dom_pft = min(self.dom_pft)
+            max_dom_pft = max(self.dom_pft)
+
+            #-- check dom_pft values should be between 1-78
+            if min_dom_pft <1 or max_dom_pft >MAX_PFT:
```

Comment:
- [ ] 0 is actually allowed as it's bare soil. The unit test should also be changed for this as well.

Reply:
This is now changed to 1, so the user has the capability to choose `--dompft 0`. 
Thanks for your comment. 
https://github.com/ESCOMP/CTSM/pull/1461#discussion_r802309262